symfit/core/support.py

```diff
@@ -357,8 +375,18 @@ def wrapped_func(*args, **kwargs):
         return wrapped_func
 
 
-class D(sympy.Derivative):
+def D(*args, **kwargs):
+    return sympy.Derivative(*args, **kwargs)
```

Comment:
Why the change?

Reply:
In essence, I wanted to give these derivatives a name property. But because of the way sympy works, the `__new__` method creates a new `Expr`, whatever you do. So the simple inheritance doesn't work. In the old way of doing `class D(sympy.Derivative)`, with a name property on it, the resulting object is still a `sympy.Derivative`, not a `D`. 

Maybe I should spend some more time trying to understand exactly how this works to see if we can change it, but to get on with the rest of the code development I choose this solution.

Reply:
I was afraid of something like that. Put a TODO or FIXME comment here for now then.